### PR TITLE
Add assets to the dewp distribution as it is needed.

### LINKED ISSUE #46748: `@woocommerce/dependency-extraction-webpack-plugin` version 2.4.0 is missing files
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The `@woocommerce/dependency-extraction-webpack-plugin` packages on NPM looks to be missing some files.  
It can be seen on the npm repository here:

- v2.4.0 - https://www.npmjs.com/package/@woocommerce/dependency-extraction-webpack-plugin/v/2.4.0?activeTab=code
- v2.3.0 - https://www.npmjs.com/package/@woocommerce/dependency-extraction-webpack-plugin/v/2.3.0?activeTab=code

### Expected behavior

Using the released version of the `@woocommerce/dependency-extraction-webpack-plugin` works correctly.

### Actual behavior

An error is return when the `@woocommerce/dependency-extraction-webpack-plugin` is used in a webpack configuration (example uses wp-scripts):
```
$ wp-scripts build
[webpack-cli] Failed to load '/path/to/project/webpack.config.js' config
[webpack-cli] Error: Cannot find module '../assets/packages'
Require stack:
- /path/to/project/node_modules/@woocommerce/dependency-extraction-webpack-plugin/src/index.js
- /path/to/project/webpack.config.js
```

### Steps to reproduce

1. Create in `package.json`
2. Install `@wordpress/scripts` and `@woocommerce/dependency-extraction-webpack-plugin` as dependencies
3. Create a `webpack.config.js` like:
    ```
    const defaultConfig = require('@wordpress/scripts/config/webpack.config');
    const WooCommerceDependencyExtractionWebpackPlugin = require('@woocommerce/dependency-extraction-webpack-plugin');
    
    module.exports = {
        ...defaultConfig,
        entry: {
            'plugin': './resources/js/plugin',
        },
        output: {
            publicPath: './',
            path: __dirname + '/assets/js',
            filename: '[name].js',
        },
        plugins: [
            ...defaultConfig.plugins.filter(
                (plugin) =>
                    plugin.constructor.name !== 'DependencyExtractionWebpackPlugin'
            ),
            new WooCommerceDependencyExtractionWebpackPlugin()
        ]
    };
```
4. Run `npm install` and `npm run build`
5. Observe error message being returned

### WordPress Environment

```
...
    "devDependencies": {
        "@woocommerce/dependency-extraction-webpack-plugin": "2.4.0",
        "@wordpress/scripts": "26.19.0"
    },
...
```

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

This should resolve https://github.com/woocommerce/woocommerce/issues/46748

A new release will need to be done.

### How to test the changes in this Pull Request:

1. Run `pnpm pack` from this package's directory
2. Uncompress the tarball
3. You should see assets folder bundled

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Fix a bug where the assets folder was not distributed with @woocommerce/dependency-extraction-webpack-plugin

#### Comment 

</details>

### CODE REVIEW

**Review by jonathansadowski — APPROVED** [review]
Thanks for the fix here @samueljseay!

For posterity, before:
```
tar -tf woocommerce-dependency-extraction-webpack-plugin-2.4.0.tgz
package/src/index.js
package/README.md
package/package.json
```

After:
```
tar -tf woocommerce-dependency-extraction-webpack-plugin-2.4.1.tgz
package/src/index.js
package/assets/packages.js
package/README.md
package/package.json
```

File: packages/js/dependency-extraction-webpack-plugin/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@woocommerce/dependency-extraction-webpack-plugin",
-	"version": "2.4.0",
+	"version": "2.4.1",
```
Comment (psealock) [inline packages/js/dependency-extraction-webpack-plugin/package.json]: No need to bump versions. The package release scripts handle all version bumping along with changelog compilation tasks.

Comment (samueljseay) [reply packages/js/dependency-extraction-webpack-plugin/package.json]: Thanks 🙏🏻

**Comment by rrennick** [comment]
@samueljseay Package release PRs need to be left open for the package release workflow to work. Once the branch has been merged, the tool logs 

```
Performing dry run of @woocommerce/dependency-extraction-webpack-plugin... @woocommerce/dependency-extraction-webpack-plugin does not have anything to update.
```

**Comment by psealock** [comment]
@rrennick Thats not my understanding.

You need to prepare the packages first with this Action https://github.com/woocommerce/woocommerce/actions/workflows/prepare-package-release.yml

Next, you can run the actual release action https://github.com/woocommerce/woocommerce/actions/workflows/package-release.yml

Full details in https://github.com/woocommerce/woocommerce/tree/trunk/tools/package-release

**Comment by psealock** [comment]
3.0.0 Published to NPM https://www.npmjs.com/package/@woocommerce/dependency-extraction-webpack-plugin. The major version bump was due to a previously unpublished major change, see full PR [here](https://github.com/woocommerce/woocommerce/pull/46847).

**Pull request merged**